### ISSUE TITLE
Having an issue when not specifying letsencrypt_pause_services

### ISSUE THREAD

**Issue by dkniffin** [jaywink/ansible-letsencrypt#20]
I'm trying to use this role to set up letsencrypt, and I'm getting the following error:

```
TASK: [jaywink.letsencrypt | Starting paused Services] ************************
fatal: [104.131.48.189] => with_items expects a list or a set
```

Looking through the code, [this](https://github.com/jaywink/ansible-letsencrypt/blob/master/tasks/cert.yaml#L31) appears to be the line it's failing on, so `_services_stopped.results` is returning something unexpected.

Looking further, it appears that variable is set [here](https://github.com/jaywink/ansible-letsencrypt/blob/master/tasks/cert.yaml#L14), but in my case, that step is skipped:

```
TASK: [jaywink.letsencrypt | Stopping Services] *******************************
skipping: [104.131.48.189]
```

I'm not quite sure how to solve this. Maybe we need to add something to the `when` line for the `Starting paused Services` step, to check if there's anything to be started back up?

**Comment by jaywink** [comment]
Any idea what `_services_stopped.results` contains? If you can still reproduce this, would you mind adding before `-name: Starting paused Services` task a printout like:

```
- debug: var=_services_stopped
```

That should give us a clue what it contains. Still, the role should be able to work around this, so we need to make sure the whole starting services is skipped if the stopping failed. If it's just undefined this would do:

```
  with_items: "{{ _services_stopped.results|default([]) }}"
```

Assuming `_services_stopped.results` contains something else, we need to check if it's a list or set :P

Thanks for reporting!

**Comment by jaywink** [comment]
OK I just re-read and noticed you said stopping services is skipped :) I wonder why? We don't have a `when` there at all and we force an empty list as default. Could you tell what you have in your variables for this? Which version of this role do you have and which version of Ansible?

**Comment by dkniffin** [comment]
Alright, I added the debug line:

```
TASK: [jaywink.letsencrypt | debug var=_services_stopped] *********************
<104.131.48.189> ESTABLISH CONNECTION FOR USER: root
ok: [104.131.48.189] => {
    "var": {
        "_services_stopped": {
            "changed": false,
            "skipped": true
        }
    }
}
```

**Comment by dkniffin** [comment]
For variables, I have `letsencrypt_domain` and `letsencrypt_email` set to my domain and email. I'm running the latest version of this role (0.5.0), and ansible 1.9.3

**Comment by dkniffin** [comment]
I also just ran it again with `debug var=letsencrypt_pause_services` before the `Stopping Services` and got:

```
TASK: [jaywink.letsencrypt | debug var=letsencrypt_pause_services] ************
<104.131.48.189> ESTABLISH CONNECTION FOR USER: root
ok: [104.131.48.189] => {
    "var": {
        "letsencrypt_pause_services": []
    }
}
```

But it's still skipping the `Stopping Services`

**Comment by jaywink** [comment]
Would adding a default to the starting services help? I honestly don't know why it's skipping the stopping services, have to look into it. I tried myself with an empty list (not defined at all in my own vars) and it ran properly, though with Ansible 2.x.

Could you try modifying `with_items` in the starting services to this and reporting back?

```
with_items: "{{ _services_stopped.results|default([]) }}"
```

If that helps I can add that since it shouldn't at least cause any other harm.

**Comment by dkniffin** [comment]
It looks like that fixed it. Now I'm running into a different issue:

```
TASK: [jaywink.letsencrypt | Obtain or renew cert for domain] *****************
<104.131.48.189> ESTABLISH CONNECTION FOR USER: root
<104.131.48.189> REMOTE_MODULE command chdir=/opt/certbot executable=/bin/bash ./certbot-auto certonly --text -n --no-self-upgrade -m  --domains tape-letsencrypt-testing.sboxes.co,www.tape-letsencrypt-testing.sboxes.co --agree-tos --standalone --expand [ ' - - r e n e w - b y - d e f a u l t ' ] 2>&1 #USE_SHELL
<104.131.48.189> EXEC ssh -C -tt -vvv -o ForwardAgent=yes -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 104.131.48.189 /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1479210888.67-253342816770440 && echo $HOME/.ansible/tmp/ansible-tmp-1479210888.67-253342816770440'
<104.131.48.189> PUT /var/folders/db/rylnjwqx6m596g9mx83v_g_r0000gn/T/tmpUohwGu TO /root/.ansible/tmp/ansible-tmp-1479210888.67-253342816770440/command
<104.131.48.189> EXEC ssh -C -tt -vvv -o ForwardAgent=yes -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=root -o ConnectTimeout=10 104.131.48.189 /bin/sh -c 'LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1479210888.67-253342816770440/command; rm -rf /root/.ansible/tmp/ansible-tmp-1479210888.67-253342816770440/ >/dev/null 2>&1'
failed: [104.131.48.189] => {"changed": true, "cmd": "./certbot-auto certonly --text -n --no-self-upgrade -m derek.kniffin@smashingboxes.com --domains tape-letsencrypt-testing.sboxes.co,www.tape-letsencrypt-testing.sboxes.co --agree-tos --standalone --expand [ ' - - r e n e w - b y - d e f a u l t ' ] 2>&1", "delta": "0:01:44.747313", "end": "2016-11-15 11:56:35.862111", "rc": 2, "start": "2016-11-15 11:54:51.114798", "warnings": []}
stdout: Bootstrapping dependencies for Debian-based OSes...
Hit:1 http://security.ubuntu.com/ubuntu xenial-security InRelease
Hit:2 http://apt.postgresql.org/pub/repos/apt xenial-pgdg InRelease
Hit:3 http://nyc2.mirrors.digitalocean.com/ubuntu xenial InRelease
Hit:4 http://nyc2.mirrors.digitalocean.com/ubuntu xenial-updates InRelease
Hit:5 http://nyc2.mirrors.digitalocean.com/ubuntu xenial-backports InRelease
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
augeas-lenses is already the newest version (1.4.0-0ubuntu1).
augeas-lenses set to manually installed.
ca-certificates is already the newest version (20160104ubuntu1).
gcc is already the newest version (4:5.3.1-1ubuntu1).
gcc set to manually installed.
libaugeas0 is already the newest version (1.4.0-0ubuntu1).
libffi-dev is already the newest version (3.2.1-4).
python is already the newest version (2.7.11-1).
python set to manually installed.
python-dev is already the newest version (2.7.11-1).
dialog is already the newest version (1.3-20160209-1).
python-virtualenv is already the newest version (15.0.1+ds-3).
virtualenv is already the newest version (15.0.1+ds-3).
virtualenv set to manually installed.
libssl-dev is already the newest version (1.0.2g-1ubuntu4.5).
libssl-dev set to manually installed.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Creating virtual environment...
Installing Python packages...
Installation succeeded.
usage:
  certbot-auto [SUBCOMMAND] [options] [-d domain] [-d domain] ...

Certbot can obtain and install HTTPS/TLS/SSL certificates.  By default,
it will attempt to use a webserver both for obtaining and installing the
cert. Major SUBCOMMANDS are:

  (default) run        Obtain & install a cert in your current webserver
  certonly             Obtain cert, but do not install it (aka "auth")
  install              Install a previously obtained cert in a server
  renew                Renew previously obtained certs that are near expiry
  revoke               Revoke a previously obtained certificate
  register             Perform tasks related to registering with the CA
  rollback             Rollback server configuration changes made during install
  config_changes       Show changes made to server config during installation
  plugins              Display information about installed plugins
letsencrypt: error: unrecognized arguments: [  - - r e n e w - b y - d e f a u l t  ]
...ignoring
```

**Comment by jaywink** [comment]
I've added the default to stopped services as talked above but it looks like the other issue is due to Ansible 1.9.3 - I managed to replicate it with that version just as yours above. Unfortunately, with the master branch of this role it fails even earlier due to some refactoring that has been done which will be out in the next release.

Unfortunately I don't have time to maintain this role for old Ansible versions which would require not following Ansible 2 documentation due to changes they have made. Ansible 2 has already been out for quite some time, I suggest if possible update to that, if possible.

Sorry, closing this issue as the original problem was worked around and the last problem is due to old Ansible version. Will update the role to say supported Ansible is 2.x only.
